### ISSUE TITLE
UI Shell: Side navigation rail lacks focus indicator in Windows High Contrast Mode

### ISSUE THREAD

**Issue by kinueng** [carbon-design-system/carbon#7313]
## Environment

> Operating system

Windows 10

> Browser

Firefox 83.0 (64-bit)

> Assistive technology used to verify

Windows High Contrast Mode 
(Left Alt + Left Shift + PrtSc)

## Detailed description

> What version of the Carbon Design System are you using?

I used the website to test, Carbon Components React v7.24.0 at the time of this writing.
https://react.carbondesignsystem.com/?path=/story/ui-shell--side-nav-rail-w-header

My specific usage: 
```
<SideNav ...>
     <SideNavItems .../>
          <SideNavLink .../>
...
```

> What did you expect to happen?

When using the keyboard to tab to each of the menu items in the left side vertical menu, I expected some kind of visual indicator on which item was in focus.  A previous issue (https://github.com/carbon-design-system/carbon/issues/6897) reported this problem and was closed as fixed, but I am unable to verify the fix on the Carbon React live demo website.

> What happened instead?

When tabbing to the side navigation menu, the menu horizontally expands to show the icon with its corresponding label.  I expected to see the menu expand and see a focus indicator on the top most menu item.  Tabbing through the menu items also do not show a visual indicator on which item is in focus.

> What WCAG 2.1 checkpoint does the issue violate?

https://www.ibm.com/able/guidelines/ci162/focus_visible.html

## Additional information

On Edge, the focus visual indicator is the removal of the white border from the menu item.

**Comment by tw15egan** [comment]
This is a Firefox-specific issue, Edge in HCM shows proper focus colors. FIrefox's `prefers-contrast` flag is new, but I'm working on adding focus styles in a PR that will be available shortly
